### DISCUSSION TITLE
how to keep sessions between request

### DISCUSSION THREAD

**Discussion by danimera** [mojolicious/mojo#1939] [Q&A] [answered]
if i writte $c->session->{foo} = 'bar';
in my next request foo is empty :(

any example to work with sessions ?

**Comment by daleif** [comment]
You probably need to provide more of an example. I just added your example to one of my development setups and it remembers `foo` just fine.

Though normally I set it using `$c->session( foo => 'bar' )` but both works.

**Comment by ghandmann** [comment, marked as answer]
I quickly hacked together a lite app and it just works as documented in [Mojolicous::Controller](https://metacpan.org/pod/Mojolicious::Controller#session):

```perl
#!/usr/bin/env perl
use Mojolicious::Lite -signatures;

get '/session/create' => sub($c) {
	$c->session->{foo} = "bar";
	$c->render(text => "ok");
};

get '/session/read' => sub($c) {
	$c->render(text => "foo from session=" . $c->session->{foo});
};

app->start;
```
